cpp/src/arrow/dataset/dataset_writer_test.cc

```diff
@@ -206,6 +206,14 @@ TEST_F(DatasetWriterTestFixture, Basic) {
   AssertCreatedData({{"testdir/chunk-0.arrow", 0, 100}});
 }
 
+TEST_F(DatasetWriterTestFixture, BasicFilePrefix) {
```

Comment:
Can you add a test that uses both a directory and a prefix?  We don't have any partitioning schemes (yet) that can generate this sort of thing but it should be perfectly valid as far as this class is concerned.

Reply:
Added the test.